pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiLanguages.scala

```diff
@@ -22,30 +20,49 @@ object TeiLanguages {
     *          <msContents>
     *            <textLang mainLang={id} source="IANA">{label}</textLang>
     *
+    * This function extracts all the nodes from a parsed XML and returns
+    * a list of (id, label) pairs.
     */
-  def apply(xml: Elem): Result[List[Language]] = {
-    val nodes = (xml \\ "msDesc" \ "msContents" \ "textLang").toList
-
-    val eitherLanguages = nodes.map { n =>
-      val label = n.text
-      val mainLangId = (n \@ "mainLang").toLowerCase
-      val otherLangId = (n \@ "otherLangs").toLowerCase
-      val langId = (mainLangId, otherLangId) match {
-        case (id1, id2) if id2.isEmpty && id1.nonEmpty => Right(id1)
-        case (id1, id2) if id1.isEmpty && id2.nonEmpty => Right(id2)
-        case (id1, id2) if id2.isEmpty && id1.isEmpty =>
-          Left(
-            new RuntimeException(
-              s"Cannot find a language id in ${n.toString()}"
-            )
-          )
-        case _ =>
-          Left(
-            new RuntimeException(s"Multiple language ids in ${n.toString()}")
-          )
+  def findNodes(xml: Elem): Try[Seq[(String, String)]] =
+    Try {
+      (xml \\ "msDesc" \ "msContents" \ "textLang").flatMap { n =>
+        val label = n.text
+
+        val mainLangId = (n \@ "mainLang").toLowerCase
+        val otherLangId = (n \@ "otherLangs").toLowerCase
+
+        val langId = (mainLangId, otherLangId) match {
+          case (id1, id2) if id2.isEmpty && id1.nonEmpty => Some(id1)
+          case (id1, id2) if id1.isEmpty && id2.nonEmpty => Some(id2)
+          case (id1, id2) if id2.isEmpty && id1.isEmpty =>
+            throw new RuntimeException(s"Cannot find a language ID in $n")
+            None
```

Comment:
This is unreachable code. It's no longer possible for this block to return None, either here or below, I think? I think you don't need options at all here